quipucords/scanner/openshift/api.py

```diff
@@ -138,6 +139,40 @@ def can_connect(self, raise_exception=False, **kwargs):
             return False
         return True
 
+    # Adding support for fetching OCP Prometheus Metrics
+    # Reference: https://access.redhat.com/solutions/3775611
+
+    @catch_k8s_exception
+    def metrics_query(self, query):
+        """Execute an OpenShift Prometheus Query."""
+        kube_config = self._configuration
+        metrics_host = self._metrics_host()
+        if not metrics_host:
+            return []
+        response = self._api_client.request(
+            "GET",
+            url=f"https://{metrics_host}/api/v1/query?query={query}",
```

Comment:
Will the `query` value always be _extremely_ simple, or should we URL encode it?

My hunch is that we should either URL encode it, _or_ better yet, use the optional `query_params` argument for `request`:

```
>>> from kubernetes.client import ApiClient
>>> help(ApiClient.request)
Help on function request in module kubernetes.client.api_client:

request(self, method, url, query_params=None, headers=None, post_params=None, body=None, _preload_content=True, _request_timeout=None)
    Makes the HTTP request using RESTClient.
```

If you want to check how that argument works, I think this class is defined here: https://github.com/kubernetes-client/python/blob/76ec6f5ce5572f66d82717e296480dc42584f0b7/kubernetes/client/api_client.py#L368

And that look like a thin wrapper for this: https://github.com/kubernetes-client/python/blob/76ec6f5ce5572f66d82717e296480dc42584f0b7/kubernetes/client/rest.py#L113